scanpy/preprocessing/_simple.py

```diff
@@ -754,7 +755,9 @@ def scale(
         raise ValueError(f"`layer` argument inappropriate for value of type {type(X)}")
     if obsm is not None:
         raise ValueError(f"`obsm` argument inappropriate for value of type {type(X)}")
-    return scale_array(X, zero_center=zero_center, max_value=max_value, copy=copy)
+    return scale_array(
+        X, zero_center=zero_center, max_value=max_value, copy=copy, mask=mask
+    )
```

Comment:
Hi, I have a quick question about the scale function. If I understood correctly how `singledispatch` works, this part of the code would only run when the data given is not an `np.ndarray`, nor a `sparse matrix`, nor `anndata` (because of the later defined functions). For what situation would this main part run for which it would be beneficial? I tried and I can see for example for a string argument it would only fail inside the `scale_array` function. Is it useful if someone gives an array-type object so it would still compute the scaling?

Reply:
The main part runs for anything that *doesn’t* have a registered function, it’s the fallback case.